File: employee_frontend/lib/models/employee.dart
class Employee {
  final String id;
  final int employeeId; // Add this field
  final String name;
  final String rating;
  final String suggestedNewRating;

  Employee({
    required this.id,
    required this.employeeId, // Add this field
    required this.name,
    required this.rating,
    required this.suggestedNewRating,
  });

  // Factory constructor to create Employee from JSON
  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      id: json['_id'] ?? '',
      employeeId: json['employeeId'] ?? 0, // Parse employeeId
      name: json['name'] ?? '',
      rating: json['rating'] ?? '',
      suggestedNewRating: json['suggestedNewRating'] ?? '',
    );
  }
}

File: EmployeeBackend/src/main/java/com/estuate/employeeperformance/model/Employee.java
package com.estuate.employeeperformance.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
//Lombok is a Java library that helps to reduce boilerplate code by using simple annotations.

@Data // automatically generates getters and setters
@AllArgsConstructor // Generates a constructor with all the fields as parameters.
@NoArgsConstructor // Generates a no-argument constructor (default constructor).
@Document(collection = "employees") // Maps this class to the employees collection in MongoDB.
public class Employee {
    @Id // primary identifier
    private String id; // mongodb assigned ID
    private int employeeId; // Standard Employee ID
    private String name;
    private String rating;
    private String suggestedNewRating;
}
